app/models/storage_tables/blob.rb

```diff
@@ -33,8 +33,15 @@ def checksum
     def destroy!
       raise StorageTables::ActiveRecordError, "Cannot delete blob attached to a record" if attachments_count.positive?
 
-      service.delete(checksum)
-      super
+      version = service.delete(checksum)
+
+      begin
+        super
+      rescue StandardError
+        service.restore(checksum, version:) if version.respond_to?(:delete_marker)
```

Comment:
De huidige implementatie is behoorlijk veel gebaseerd op die van S3, zou nog wat meer generiek kunnen wat mij betreft.

```suggestion
      delete_result = service.delete(checksum)

      begin
        super
      rescue StandardError
        service.restore(checksum, delete_result)
```